### PR TITLE
PR: Add teletubbies page

### PR DESCRIPTION
**Pull request by naidendev**
- Add Teletubbies layout
- Implement search Teletubbies by name 
- Remove double quote

### CODE REVIEW

**Comment by star95417** [comment]
Looks good, here's some of the code you need to fix.
- Most of commit name are good to see but fix this one.
   `render each Teletubby card`
    How about `add Telebby card`?

File: package.json [hunk 1]
```diff
@@ -5,13 +5,15 @@
...
     "@emotion/styled": "^11.11.0",
+    "@material-ui/core": "^4.12.4",
     "@mui/icons-material": "^5.11.16",
     "@mui/material": "^5.13.4",
     "@mui/styled-engine-sc": "^5.12.0",
     "@testing-library/jest-dom": "^5.16.5",
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
     "axios": "^1.4.0",
+    "bootstrap": "^5.3.0",
```
Comment (star95417) [inline package.json:16]: I don't think we need to install `bootstrap` in this project. Please remove it.

File: src/App.css [hunk 2]
```diff
@@ -36,3 +36,6 @@
     transform: rotate(360deg);
   }
 }
+.teletubby-grid-container > div {
+  border: 1px solid black;
```
Comment (star95417) [inline src/App.css:40]: I don't think we need to add custom style in here, please find another way by using origin MUI components and it's attribute.

File: src/App.js [hunk 3]
```diff
@@ -2,8 +2,10 @@ import { BrowserRouter, Routes, Route } from "react-router-dom";
 
 import Layout from "pages/Layout";
 import Home from "pages/Home";
-import NoPage from "pages/NoPage";
+
 import "./App.css";
+import Teletubbies from "pages/Teletubbies/index";
```
Comment (star95417) [inline src/App.js:7]: "pages/Teletubbies/index" is okay but I suggest to fix it as "pages/Teletubbies".
And it will work too.

File: src/App.js [hunk 4]
```diff
@@ -2,8 +2,10 @@ import { BrowserRouter, Routes, Route } from "react-router-dom";
 
 import Layout from "pages/Layout";
 import Home from "pages/Home";
-import NoPage from "pages/NoPage";
```
Comment (star95417) [inline src/App.js:5]: Why did move this line to bottom? Don't change any line that is not relate your feature.

File: src/api/index.js [hunk 5]
```diff
@@ -18,3 +18,18 @@ export const getCurrentWeatherData = async () => {
     throw error;
   }
 };
+
+const teletubbyOptions = {
+  method:"GET",
+  url:"./teletubbies.json",
```
Comment (star95417) [inline src/api/index.js:24]: `url:"./teletubbies.json"` will work but `url:"/teletubbies.json"` makes sense for me

File: src/api/teletubbies.json
Comment (star95417) [inline src/api/teletubbies.json:1]: I don't think we need this file in this project

File: src/pages/Teletubbies/index.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
+import React, { useState, useRef, useEffect } from "react";
+import { Grid, TextField, Typography } from "@mui/material"
+import { getCurrenTeletubbyData } from 'api/index';
```
Comment (star95417) [inline src/pages/Teletubbies/index.js:3]: `from 'api/index'` also work but `from 'api'` is better in this case.

Comment (naidendev) [reply src/pages/Teletubbies/index.js:3]: Thank for your help! I changed it to 'api'

File: src/api/teletubbies.json
Comment (naidendev) [reply src/api/teletubbies.json:1]: Yes. I removed this JSON file.

File: src/api/index.js (hunk 5 above)
Comment (naidendev) [reply src/api/index.js:24]: Yes, I changed this route.

File: src/App.js (hunk 4 above)
Comment (naidendev) [reply src/App.js:5]: Yes. I will refer you do it.

File: src/App.js (hunk 3 above)
Comment (naidendev) [reply src/App.js:7]: That's a good idea!

File: package.json (hunk 1 above)
Comment (naidendev) [reply package.json:16]: I removed this.

**Comment by star95417** [comment]
Make sure your PR name follow the convention

File: src/pages/Teletubbies/index.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    } catch (error) {
+      console.error(error);
+    }
+  }
+  
+  useEffect(() => {
+    getTeletubbies();
+  }, []);
+
+  //Render each Teletubby card
```
[the remark this answers is not in the document]
Comment (naidendev) [reply src/pages/Teletubbies/index.js:21]: I do it.

File: src/pages/Teletubbies/index.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+          marginRight: "auto",
+        }}
+        mt={4}
+        mb={3}
+      >
+        Teletubbies
+      </Typography>
+
+      <Grid container spacing={3} className="teletubby-grid-container">
+        {teletubbies.map((teletubby, index) => {
```
[the remark this answers is not in the document]
Comment (naidendev) [reply src/pages/Teletubbies/index.js:61]: Wonderful!
I changed this section.
Thank you.

File: src/pages/Teletubbies/index.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+                {trait}
+              </button>
+            ))}
+          </div>
+        </div>
+      </Grid>
+    );
+  };
+  
+  return (
```
[the remark this answers is not in the document]
Comment (naidendev) [reply src/pages/Teletubbies/index.js:45]: I remove it

File: src/pages/Teletubbies/index.js [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+          width: "50%",
+          marginLeft: "auto",
+          marginRight: "auto",
+        }}
+        mt={4}
+        mb={3}
+      >
+        Teletubbies
+      </Typography>
+      <p>s;lk;lkfj;lskdfl;ke;lfe</p>
```
Comment (star95417) [inline src/pages/Teletubbies/index.js:60]: What is this for?

Comment (naidendev) [reply src/pages/Teletubbies/index.js:60]: I deleted it.

**Comment by star95417** [comment]
I don't think you are trying to follow the convention when you commit. Please be consistent.
image

File: src/components/Teletubby/index.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+        <img
+          src={teletubby.image}
+          alt={teletubby.name}
+          style={{ width: "50%" }}
+        />
+        <div style={{ padding: "10px" }}>
+          <h3>{teletubby.name}</h3>
+          <p>{teletubby.description}</p>
+          <div>
+            {}
```
Comment (star95417) [inline src/components/Teletubby/index.js:16]: What is this for?

**Comment by star95417** [comment]
- Keep the header at the top even user scrolldown. (You can ask the other teammates)  
image

- Provide the padding left on your page.
image

- Please remove these warnings.
   image

File: src/App.js [hunk 12]
```diff
@@ -3,16 +3,17 @@ import { BrowserRouter, Routes, Route } from "react-router-dom";
 import Layout from "pages/Layout";
 import Home from "pages/Home";
 import NoPage from "pages/NoPage";
+import Teletubbies from "pages/Teletubbies";
 import Buildings from "pages/Buildings";
 import "./App.css";
-
```
Comment (star95417) [inline src/App.js:8]: Don't remove this

File: src/api/index.js [hunk 13]
```diff
@@ -1,12 +1,12 @@
-import axios from 'axios';
```
Comment (star95417) [inline src/api/index.js:1]: Check your prettier settings and convert all the double quote to single quote

File: src/pages/Layout/index.js [hunk 14]
```diff
@@ -1,16 +1,25 @@
 import { Outlet } from "react-router-dom";
+import { Box } from "@mui/material";
 
 import Header from "components/Header";
 
 const Layout = () => {
   return (
-    <>
+    <Box
+    // sx={{
```
Comment (star95417) [inline src/pages/Layout/index.js:9]: remove unnecessary commented code

File: src/pages/Layout/index.js [hunk 15]
```diff
@@ -1,16 +1,25 @@
...
+    //   flexDirection: "column",
+    //   overflow: "hidden",
+    //   height: "100vh",
+    // }}
+    >
       <header>
         <Header />
       </header>
 
+      {/* <Box sx={{ overflow: "auto" }}> */}
```
Comment (star95417) [inline src/pages/Layout/index.js:20]: same here

**Comment by star95417** [comment]
Overall UI looks good to me, BTW can you fix these thing?
- remove the warring.
   image

- Can you make this image height as like the component?
image

File: src/pages/Layout/index.js (hunk 14 above)
Comment (naidendev) [reply src/pages/Layout/index.js:9]: I deleted  unnecessary commented code

File: src/pages/Layout/index.js (hunk 15 above)
Comment (naidendev) [reply src/pages/Layout/index.js:20]: I deleted  unnecessary commented code

File: src/api/index.js (hunk 13 above)
Comment (naidendev) [reply src/api/index.js:1]: convert it

File: src/App.js (hunk 12 above)
Comment (naidendev) [reply src/App.js:9]: yes.

**Pull request merged**